1xx-enable-ssl-for-kafka-bridge.md

```diff
@@ -0,0 +1,110 @@
+# Support TLS/SSL on the HTTP interface 
+
+This proposal adds an option to enable TLS encryption on HTTP interface so that requests from clients to the Kafka Bridge can be sent with encryption.
+
+## Current situation
+
+Strimzi does not support TLS/SSL for HTTP Bridge, so clients connections to HTTP Bridge is not encrypted at all.
+
+## Motivation
+
+The connection between HTTP Bridge and Kafka cluster can be secured however, currently there is no support for securing the connections to HTTP Bridge from clients. The current recommendation is to use firewalls or API gateways to secure the client connections. This requires users to make additional configurations and does not provide the best user experience.  
+
+## Proposal
+
+This proposal adds new configurations to enable SSL server and specify locations of certificate and key files in PEM format so that client connections to HTTP Bridge are encrypted. The new configurations will be added to the current `http` prefixed configurations:
+- http.ssl.enable
+- http.ssl.port
+- http.ssl.keystore.location
+- http.ssl.keystore.key.location
+- http.ssl.enabled.protocols
+- http.ssl.enabled.cipher.suites
+
+When `http.ssl.enable` is set to true, HTTP Bridge server will be started with SSL enabled and the key and certificate loaded from the locations defined by `http.ssl.keystore.location` and `http.ssl.keystore.key.location` configurations. If `http.ssl.port` configuration is not set, the server port will be set to `8443` by default. This will allow users to make TLS encrypted connections from their clients for all the HTTP Bridge endpoints for client operations. However, connections to the following endpoints for monitoring will stay unencrypted:
+- `/healthy`
+- `/ready`
+- `/metrics`
+- `/openapi`
+- `/openapiv2`
+- `/openapiv3`
+- `/info`
+
+When `http.ssl.enable` is set, these endpoints will have a separate listener, that is always PLAINTEXT. It will use the port defined by the existing `http.port` that is set to `8080` by default. The purpose of these endpoints is to be exposed only internally for the health checks, getting OpenAPI specification and metrics collection. In the future, we are likely to implement authentication and authorization for HTTP Bridge. By having separate listeners, these monitoring endpoints will not be affected by such changes.
```

Comment:
As I said before, this just postpones the issue for when authentication is introduced. I think the proposal should be more forward-thinking and use a completely separate port for the metrics and healthchecks.